### PR TITLE
returning items with the shape for shaped recipe

### PR DESCRIPTION
**Pull request by MrEnxo**
so instead of just getting the items when you try to get the ingredients of a recipe you also get the air and thus the shape of the shaped recipe

### CODE REVIEW

**Comment by ShaneBeee** [comment]
I tried this before as a test, and it only works for custom recipes.
Vanilla recipes don't actually contain air or anything like that, its just the shape on how they're done.

**Comment by MrEnxo** [comment]
thats weird... yea i have that same issue but i mean it works for now for custom recipes

**Comment by MrEnxo** [comment]
I made a fix so it works with vanilla recipes now

**Comment by MrEnxo** [comment]
@ShaneBeee

**Comment by ShaneBeee** [comment]
> @ShaneBeee

you don't need to tag me, I get these on my discord and emails as well. 



> I made a fix so it works with vanilla recipes now

Awesome, I will give it a try and see how it goes

**Comment by MrEnxo** [comment]
alright.

**Comment by MrEnxo** [comment]
just added a patch to fix things!

**Comment by MrEnxo** [comment]
currently this solution does not like any recipe with 9 items for some reason

**Comment by MrEnxo** [comment]
this is now fixed

**Comment by ShaneBeee** [comment]
this doesn't seem to work for small recipes, ie: jack o lantern (where its a 2x2 recipe)
Screenshot 2022-07-29 at 1 42 58 PM

**Comment by MrEnxo** [comment]
i mean it does get the shape but just not the right size for it... idk how we would solve this though

**Comment by ShaneBeee** [comment]
> i mean it does get the shape but just not the right size for it... idk how we would solve this though

now you know why I never wanted to touch this haha

**Comment by MrEnxo** [comment]
lolo

**Comment by ShaneBeee** [comment]
WHAT IF... you checked the size of shape?
if its 2, that means its 2x2
if its 3, that means its 3x3
yeah?

**Comment by MrEnxo** [comment]
hmmm maybe let me try that

**Comment by MrEnxo** [comment]
IT WORKS!!!!

**Comment by ShaneBeee** [comment]
good job.
Ok now... since both blocks of code are nearly identical, I feel like you could join them and just have vars set for the size
ex: `int i = size.length() + 1` <-- or some jazz like that

**Comment by MrEnxo** [comment]
ig i could but im lazy but ill do it brb

**Comment by MrEnxo** [comment]
alright done

**Comment by ShaneBeee** [comment]
much better thanks, now let me test...

File: src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java [hunk 1]
```diff
@@ -61,10 +63,26 @@ protected ItemType[] get(Event e) {
...
-                        if (ingredient == null) continue;
-                        items.add(new ItemType(ingredient));
+                    String[] shape = shapedRecipe.getShape();
+
+                    for (int i = 0; i < Math.pow(shape.length, 2); i++) {
+                        items.add(new ItemType(Material.AIR));
                     }
+
+                    for (int i = 0; i < shape.length; i++) {
+
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java:73]: Everything looks good and works, now all I need you to do is remove these excessive blank lines please and thank you :)

**Review by ShaneBeee — CHANGES_REQUESTED** [review]

File: src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java (hunk 1 above)
Comment (MrEnxo) [reply src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java]: i have fix the blank lines

File: src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java [hunk 2]
```diff
@@ -11,8 +11,10 @@
 import ch.njol.skript.lang.SkriptParser;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
+import com.shanebeestudios.skbee.SkBee;
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java:14]: This import can be removed, Im assuming it was from when you were debugging.

File: src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java [hunk 3]
```diff
@@ -11,8 +11,10 @@
...
 import org.bukkit.Bukkit;
 import org.bukkit.Keyed;
+import org.bukkit.Material;
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java:17]: I don't believe this import is being used either

File: src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java [hunk 4]
```diff
@@ -61,9 +63,16 @@ protected ItemType[] get(Event e) {
...
-                        items.add(new ItemType(ingredient));
+                    String[] shape = shapedRecipe.getShape();
+                    for (int i = 0; i < Math.pow(shape.length, 2); i++) {
+                        items.add(new ItemType(Material.AIR));
+                    }
+                    for (int i = 0; i < shape.length; i++) {
+                        for (int x = 0; x < shape[i].length(); x++) {
+                            ItemStack ingredient = shapedRecipe.getIngredientMap().get(shape[i].toCharArray()[x]);
+                            if (ingredient != null) items.set(i * shape.length + x, new ItemType(ingredient));
+
```
Comment (ShaneBeee) [inline src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java:74]: Sorry to be nitpicky but can you remove this blank line too please

File: src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java (hunk 1 above)
Comment (MrEnxo) [reply src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java]: fixed*

**Review by ShaneBeee — CHANGES_REQUESTED** [review]
Just a couple more nit picks from me. Sorry I didn't catch the two imports in the last review I did

File: src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java (hunk 4 above)
Comment (MrEnxo) [reply src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java]: fixed the imports and the line. I thought I deleted that line, I guess not. Also material is a needed import

Comment (ShaneBeee) [reply src/main/java/com/shanebeestudios/skbee/elements/recipe/expressions/ExprIngredientsOfRecipe.java]: oh yes. Material.AIR... duh ... I missed that. Thanks

**Review by ShaneBeee — APPROVED** [review]

**Comment by ShaneBeee** [comment]
merged... thank you for your contribution :)

**Pull request merged**